### ISSUE TITLE
Object properties inconsistent highlighting

### ISSUE THREAD

**Issue by cjnucette** [HerringtonDarkholme/yats.vim#83]
Hi,

In the following piece of code the `action` argument is an object:

![screenshot from 2018-09-22 10-25-13](https://user-images.githubusercontent.com/2278326/45918230-d9c51380-be51-11e8-9253-d342ab55071c.png)
As you can see it gets highlighted in a very inconsistent way. The highlighting groups I get are the following:
* in the switch statement
    ```vim
    hi<typescriptConditionalParen> trans<typescriptIdentifierName> lo<typescriptConditionalParen>
    ```
* in the array
  ```vim
  hi<typescriptFileListMethod> trans<typescriptFileListMethod> lo<typescriptFileListMethod>
  ```
* in the filter callback (Number() doesn't affect)
  ```vim
  hi<typescriptFuncCallArg> trans<typescriptProp> lo<typescriptFuncCallArg>
  ```

**Comment by tomaskallup** [comment]
Hi, sadly I think this is intended, but there probably should be a way to opt out.
What's happening is that `action.type` here get's highlighted as `typescriptDOMEventProp`, which as you can see is a special property of DOMEvents. `action.item` is as you found `typescriptFileListMethod`, again a property of FileList. Lastly the `action.index` highlight is the correct one.
Maybe there should be a documented way to disable these, or some configuration variable, as some people might like it.

I'll wait for @HerringtonDarkholme, to see what he thinks about this.

**Comment by tomaskallup** [comment]
Well not even the context matters, it's just matching keywords like `push` etc. You can see all the keywords in here https://github.com/HerringtonDarkholme/yats.vim/tree/master/syntax/yats
As far as I saw, they all link to `Keyword` so maybe we could create a new highlight to which they would link and that would then be a simple way to disable the highlight or perhaps a better solution would be to add a global option (maybe something like `disable_platform_specific_highlight`) to just completely disable loading of all those syntax files as that might also boost performance a little bit.

**Comment by HerringtonDarkholme** [comment]
Currently there is no way to disable platform specific highlight. 

I think it is good idea to disable those highlights. yats is forked from yajs which contains these highlight by default.
I wonder if users want these highlight.

**Comment by HerringtonDarkholme** [comment]
Fixed in f054d56
